File: aulas/aula.cpp
// Aula de C++ - Conceitos Fundamentais
#include <iostream>
#include <vector>
#include <set>
#include <map>
#include <cmath>
using namespace std;

// Função (fora da main)
int soma(int a, int b) {
    return a + b;
}

// Classe Pessoa
class Pessoa {
public:
    string nome;
    int idade;

    Pessoa(string n, int i) {
        nome = n;
        idade = i;
    }

    void falar() {
        cout << "Oi, eu sou " << nome << " e tenho " << idade << " anos." << endl;
    }
};

int main() {
    // 1. TIPOS DE DADOS BÁSICOS
    int inteiro = 10;
    double real = 3.14;
    string texto = "Hello";
    bool booleano = true;

    vector<int> lista = {1, 2, 3}; // vetor (lista)
    set<int> conjunto = {1, 2, 2, 3}; // set (não aceita repetição)
    map<string, int> dicionario; // map (dicionário)
    dicionario["idade"] = 17;

    cout << inteiro << " " << real << " " << texto << " " << booleano << endl;

    // ------------------------------------------------------
    // 2. OPERADORES
    int x = 5, y = 2;
    cout << x + y << endl;
    cout << x - y << endl;
    cout << x * y << endl;
    cout << x / y << endl; // divisão inteira
    cout << x % y << endl; // resto
    cout << pow(x, y) << endl; // potência

    cout << (x == y) << " " << (x > y) << " " << (x != y) << endl;

    // ------------------------------------------------------
    // 3. CONTROLE DE FLUXO
    if (x > y) {
        cout << "x é maior que y" << endl;
    } else if (x == y) {
        cout << "x é igual a y" << endl;
    } else {
        cout << "x é menor que y" << endl;
    }

    // Loop for
    for (int i = 0; i < 5; i++) {
        cout << "for: " << i << endl;
    }

    // Loop while
    int contador = 0;
    while (contador < 3) {
        cout << "while: " << contador << endl;
        contador++;
    }

    // ------------------------------------------------------
    // 4. FUNÇÕES
    cout << "Função soma: " << soma(3, 4) << endl;

    // ------------------------------------------------------
    // 5. ESTRUTURAS DE DADOS
    vector<int> numeros = {10, 20, 30};
    numeros.push_back(40);
    cout << "Lista: ";
    for (int n : numeros) cout << n << " ";
    cout << endl;

    map<string, double> aluno;
    aluno["nota"] = 9.5;
    aluno["idade"] = 16;
    cout << "Nota: " << aluno["nota"] << endl;

    // ------------------------------------------------------
    // 7. TRATAMENTO DE ERROS
    try {
        int a = 10 / 0; // isso dá erro
        cout << a << endl;
    } catch (exception& e) {
        cout << "Erro: " << e.what() << endl;
    }

    // ------------------------------------------------------
    // 8. ORIENTAÇÃO A OBJETOS
    Pessoa p1("Gui", 17);
    p1.falar();

    // ------------------------------------------------------
    // 9. MÓDULOS / BIBLIOTECAS
    cout << "Raiz quadrada de 16: " << sqrt(16) << endl;

    // ------------------------------------------------------
    // 10. PARADIGMAS
    // Imperativo
    x = 10;
    y = 20;
    cout << x + y << endl;

    // Funcional (usando lambda e transform)
    vector<int> valores = {1, 2, 3, 4};
    vector<int> dobro;
    for (int n : valores) {
        dobro.push_back(n * 2);
    }
    cout << "Dobro: ";
    for (int d : dobro) cout << d << " ";
    cout << endl;

    return 0;
}

File: aulas/aula.java
// Aula de Java - Conceitos Fundamentais

public class AulaJava {
    public static void main(String[] args) {
        // 1. TIPOS DE DADOS BÁSICOS
        int inteiro = 10;
        double real = 3.14;
        String texto = "Hello";
        boolean booleano = true;

        int[] lista = {1, 2, 3};
        java.util.HashSet<Integer> conjunto = new java.util.HashSet<>();
        conjunto.add(1);
        conjunto.add(2);
        conjunto.add(2);
        conjunto.add(3);

        java.util.HashMap<String, Object> dicionario = new java.util.HashMap<>();
        dicionario.put("nome", "Gui");
        dicionario.put("idade", 17);

        System.out.println(inteiro + " " + real + " " + texto + " " + booleano);
        System.out.println(java.util.Arrays.toString(lista));
        System.out.println(conjunto);
        System.out.println(dicionario);

        // ------------------------------------------------------
        // 2. OPERADORES
        int x = 5, y = 2;
        System.out.println(x + y);
        System.out.println(x - y);
        System.out.println(x * y);
        System.out.println(x / y);
        System.out.println(x % y);
        System.out.println(Math.pow(x, y));

        System.out.println(x == y);
        System.out.println(x > y);
        System.out.println(x != y);

        // ------------------------------------------------------
        // 3. CONTROLE DE FLUXO
        if (x > y) {
            System.out.println("x é maior que y");
        } else if (x == y) {
            System.out.println("x é igual a y");
        } else {
            System.out.println("x é menor que y");
        }

        // Loop for
        for (int i = 0; i < 5; i++) {
            System.out.println("for: " + i);
        }

        // Loop while
        int contador = 0;
        while (contador < 3) {
            System.out.println("while: " + contador);
            contador++;
        }

        // ------------------------------------------------------
        // 4. FUNÇÕES
        System.out.println("Função soma: " + soma(3, 4));

        // ------------------------------------------------------
        // 5. ESTRUTURAS DE DADOS
        java.util.ArrayList<Integer> numeros = new java.util.ArrayList<>();
        numeros.add(10);
        numeros.add(20);
        numeros.add(30);
        numeros.add(40);
        System.out.println("Lista: " + numeros);

        java.util.HashMap<String, Object> aluno = new java.util.HashMap<>();
        aluno.put("nome", "Ana");
        aluno.put("nota", 9.5);
        System.out.println("Nome: " + aluno.get("nome"));

        // ------------------------------------------------------
        // 7. TRATAMENTO DE ERROS
        try {
            System.out.println(10 / 0);
        } catch (ArithmeticException e) {
            System.out.println("Erro: divisão por zero");
        }

        // ------------------------------------------------------
        // 8. ORIENTAÇÃO A OBJETOS
        Pessoa p1 = new Pessoa("Gui", 17);
        p1.falar();

        // ------------------------------------------------------
        // 9. MÓDULOS / BIBLIOTECAS
        System.out.println("Raiz quadrada de 16: " + Math.sqrt(16));

        // ------------------------------------------------------
        // 10. PARADIGMAS
        // Imperativo
        x = 10;
        y = 20;
        System.out.println(x + y);

        // Funcional (Java Streams)
        java.util.List<Integer> valores = java.util.Arrays.asList(1, 2, 3, 4);
        java.util.List<Integer> dobro = valores.stream().map(n -> n * 2).toList();
        System.out.println("Dobro: " + dobro);
    }

    // Função (método estático)
    public static int soma(int a, int b) {
        return a + b;
    }
}

// Classe Pessoa
class Pessoa {
    String nome;
    int idade;

    Pessoa(String nome, int idade) {
        this.nome = nome;
        this.idade = idade;
    }

    void falar() {
        System.out.println("Oi, eu sou " + nome + " e tenho " + idade + " anos.");
    }
}
